pepper-apis/src/main/java/org/broadinstitute/ddp/elastic/participantslookup/ESParticipantsLookupService.java

```diff
@@ -0,0 +1,148 @@
+
+        var participantsStructuredResults = new ESParticipantsSearch(esClient, governedUserToProxy, participantsLookupResult)
+                .setResultMaxCount(resultsMaxCount)
+                .setEsIndex(participantsEsIndex)
+                .setFetchSource(PARTICIPANTS_STRUCTURED__INDEX__SOURCE)
+                .setQuery(query)
+                .search();
+
+        var esResultRows = mergeResults(
+                proxiesResults, participantsStructuredResults, governedUserToProxy, governedUserToProxyExtraSearch);
+
+        // if any extra-proxies found then fetch their data from index "users", add to proxiesResults, and merge results again
+        if (governedUserToProxyExtraSearch.size() > 0) {
+            esResultRows = reMergeResultsWithExtraFoundProxies(
+                    query, resultsMaxCount, governedUserToProxy, governedUserToProxyExtraSearch,
+                    usersEsIndex, proxiesResults, participantsStructuredResults);
+
+        }
+
+        participantsLookupResult.setResultRows(esResultRows);
+    }
+
+    /**
+     * Adds to results of "participants_structured" index the 'proxies' data results (taken from
+     * "users" index results).
+     */
+    private List<ParticipantsLookupResultRow> mergeResults(
+            Map<String, ESUsersIndexResultRow> usersResults,
+            Map<String, ESParticipantsStructuredIndexResultRow> participantsStructuredResults,
+            Map<String, String> governedUserToProxy,
+            Map<String, String> governedUserToProxyExtraSearch) {
+        List<ParticipantsLookupResultRow> resultList = new ArrayList<>();
+        for (var participant : participantsStructuredResults.values()) {
+            ParticipantsLookupResultRow resultRow = new ParticipantsLookupResultRow(participant);
+            var proxyGuid = governedUserToProxy.get(participant.getGuid());
+            if (proxyGuid != null) {
+                resultRow.setProxy(new ParticipantsLookupResultRowBase(usersResults.get(proxyGuid)));
+            } else if (participant.getProxies().size() > 0) {
+                governedUserToProxyExtraSearch.put(participant.getGuid(), participant.getProxies().get(0));
+            }
+            resultList.add(resultRow);
+        }
+        return resultList;
+    }
+
+    private List reMergeResultsWithExtraFoundProxies(
+            String query,
+            int resultsMaxCount,
+            Map<String, String> governedUserToProxy,
+            Map<String, String> governedUserToProxyExtraSearch,
+            String usersEsIndex,
+            Map proxiesResults,
+            Map participantsStructuredResults) throws IOException {
+        List esResultRows;
+        Map<String, ESUsersIndexResultRow> proxiesExtraResults = new ESUsersProxiesExtraSearch(esClient, governedUserToProxyExtraSearch)
+                .setResultMaxCount(resultsMaxCount)
+                .setEsIndex(usersEsIndex)
+                .setFetchSource(USERS__INDEX__SOURCE)
+                .setQuery(query)
```

Comment:
Setting the query here is confusing -- this extra search is looking up proxies by guid and not user-provided `query`.

Reply:
But user-provided query is executed in 1st step in "users" index (when searched for proxies). 
Is it correct that "participants" contains all except proxies? In that case the searching in "participants" index is enough to find everyone except proxies ( + proxies found in "users")?

Reply:
It is correct that the `participants_structured` index contains only participants -- no proxies. But that not what my comment is about. I'm saying this line here is confusing because the "proxy extra search" does not use the user-provided `query` as search term. Rather, it is looking for proxies by guid. So having `.setQuery(query)` is confusing.

Reply:
Ah, understood)
Yes, I thought about the same, but forgot to remove this setQuery().. Will remove

Reply:
Fixed